src/pages/accountsettings/AccountSettings.js

```diff
@@ -0,0 +1,112 @@
+import React, { useState, Component } from "react";
+import { Link } from "react-router-dom";
+// import "../signup/signup.css";
+
+import "../../common.css"
+
+const initialValues = {
+  email: "",
+  password: "",
+};
+
+function AccountSettings() {
+  const [component, setComponent] = useState("signup");
+
+  // States for registration
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
+  const [phoneNumber, setPhoneNumber] = useState("");
+  const [userName, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const [repeatPassword, setRepeatPassword] = useState("");
+
+  // Handling the name change
+  const handleName = (e) => {
+    setName(e.target.value);
+  };
+
+  // Handling the phone number change
+  const handlePhoneNumber = (e) => {
+    setPhoneNumber(e.target.value);
+  };
+
+  // Handling the email change
+  const handleEmail = (e) => {
+    setEmail(e.target.value);
+  };
+
+  // Handling the username change
+  const handleUserName = (e) => {
+    setUsername(e.target.value);
+  };
+
+  // Handling the password change
+  const handlePassword = (e) => {
+    setPassword(e.target.value);
+  };
+
+  // Handling the repeat password change
+  const handleRepeatPassword = (e) => {
+    setRepeatPassword(e.target.value);
+  };
```

Comment:
You have 6 handlers, doing exactly the same thing.
You could create a generic function, which accepts an argument.
Make reusable code, and avoid unnecessary repetition.